src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1083,25 +1084,39 @@ private static void getElementsFromReferences(Set<TreeReference> referencesRemai
     /**
      * meant to be called after deserialization and initialization of handlers
      *
-     * @param newInstance true if the form is to be used for a new entry interaction,
-     *                    false if it is using an existing IDataModel
+     * @param formInitializationMode Specifies the mode in which the form is being initialized:
+     *                               NEW_FORM: The form is being initialized for a new entry interaction.
+     *                               DRAFT_FORM_EDIT: The form is being edited from a saved draft.
+     *                               FINALIZED_FORM_EDIT: The form is being edited after being finalized.
      */
-    public void initialize(boolean newInstance, InstanceInitializationFactory factory) {
+    public void initialize(FormInitializationMode formInitializationMode) {
         HashMap<String, DataInstance> formInstances = getFormInstances();
         for (String instanceId : formInstances.keySet()) {
             DataInstance instance = formInstances.get(instanceId);
-            instance.initialize(factory, instanceId);
+            instance.initialize(instanceId);
         }
-        if (newInstance) {// only preload new forms (we may have to revisit
+        if (formInitializationMode == FormInitializationMode.NEW_FORM) {// only preload new forms (we may have to revisit
```

Comment:
While here, maybe clean up this comment? Maybe the revisiting was about edits, or if not we don't know what the intent was

Reply:
Done.